extensions/subscriptions/src/Job/SendReplyNotification.php

```diff
@@ -41,20 +45,45 @@ public function __construct(Post $post, $lastPostNumber)
         $this->lastPostNumber = $lastPostNumber;
     }
 
-    public function handle(NotificationSyncer $notifications)
+    public function handle(NotificationSyncer $notifications, SettingsRepositoryInterface $settings)
     {
         $post = $this->post;
         $discussion = $post->discussion;
 
-        $notify = $discussion->readers()
+        $usersToNotify = [];
+
+        $followers = $discussion->readers()
+            ->select('users.id', 'users.preferences', 'discussion_user.last_read_post_number')
             ->where('users.id', '!=', $post->user_id)
-            ->where('discussion_user.subscription', 'follow')
-            ->where('discussion_user.last_read_post_number', $this->lastPostNumber - 1)
-            ->get();
+            ->where('discussion_user.subscription', 'follow');
+
+        $followers->chunk(150, function (Collection $followers) use (&$usersToNotify) {
```

Comment:
I made a minor change here (diff shows I as more than it is lol) where we chunk results in case of large communities. @davwheat does that seem sensible?

Reply:
Definitely.